winforms/tests_backend/widgets/base.py

```diff
@@ -56,10 +55,19 @@ def color(self):
 
     @property
     def background_color(self):
-        if self.native.BackColor == Color.Transparent:
-            return TRANSPARENT
-        else:
-            return toga_color(self.native.BackColor)
+        return (
+            toga_color(self.native.BackColor),
+            toga_color(self.impl.interface.parent._impl.native.BackColor),
+            (
+                # self.impl.interface.style.background_color can be None or TRANSPARENT
+                # and so there will be no alpha value on them. In such cases return 0
+                # as the original alpha value.
+                self.impl.interface.style.background_color.a
+                if getattr(self.impl.interface.style.background_color, "a", None)
+                is not None
+                else 0
```

Comment:
This seem overly complicated. `background_color` is a Toga color. If it's an RGB declaration, it *must* have an  alpha (it will default to 1). If it's a HSL definition, there's a `.rgba` property to convert to RGB format (but for testbed purposes, we don't use HSL, so this is a moot point). That means the value should be either `TRANSPARENT`, or it will have an alpha value - so `getattr(bg, "a", 0)` (or `0 if bg == TRANSPARENT else bg.a` if you want to be more explicit) should be all you need here.

Reply:
Thanks, this looks much simpler.